src/internal/utils/convertSnakeToCamelCase.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Converts snake_case keys to camelCase in an object or array of objects.
+ * @param {T} obj - The object, array, or string to convert. (required)
+ * @returns {T} The converted object, array, or string.
+ */
+export function convertSnakeToCamelCase<T>(obj: T): T {
+  if (typeof obj === 'string') {
+    return obj as T;
+  }
+
+  if (Array.isArray(obj)) {
+    return obj.map((item) => convertSnakeToCamelCase(item)) as T;
+  }
+
+  if (obj && obj.constructor === Object) {
+    return Object.keys(obj).reduce((acc, key) => {
+      const camelCaseKey = toCamelCase(key);
+      // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+      (acc as any)[camelCaseKey] = convertSnakeToCamelCase<T>(
```

Comment:
do you even need to cast these?  can't ts infer that acc is still T and obj is a property of T?

Reply:
Done